### ISSUE TITLE
Virtualenv instance: login with Google is broken

### ISSUE THREAD

**Issue by shainer** [fedora-infra/anitya#437]
I downloaded the latest release of anitya. I then followed the instructions at https://github.com/release-monitoring/anitya#virtualenv to run a local instance with virtualenv.

The UI gives me the following options for login: Fedora, Google, Yahoo and "other OpenID". If I click on Google, I get redirected to localhost:5000/login/google and an error popup appears: "Error during login: The OpenID was invalid" (no credential was ever asked).

This is already puzzling because the version running at release-monitoring.org does not offer the Google login option at all. What's going wrong?

**Comment by jeremycline** [comment]
There are application settings to turn default providers on or off and release-monitoring.org is configured to turn the Google option off. My guess is because it broke and no one has fixed it yet. I see the login method is using https://www.google.com/accounts/o8/id which is now an HTTP 404. My guess is that was the OpenID 2.0 endpoint which appears to be deprecated by Google.

https://github.com/release-monitoring/anitya/pull/401 is introducing OpenID Connect so after that lands we can probably fix the Google login. I'm not as familiar with OpenID as I should be, but @puiterwijk is the expert in this area so he can probably tell me if I'm wrong :smile:

**Comment by jeremycline** [comment]
To follow up to this, I got nosy and found your blog post about using Anitya for Chakra Linux. I'm very open to accepting PRs for all the issues you noted in your post. I definitely want to drop the hard dependency on RPM and instead provide several ways to sort releases (Python, for example, has https://www.python.org/dev/peps/pep-0440/).

**Comment by pypingou** [comment]
You are correct that the Google login broke when Google dropped their OpenID endpoint and we never port it.

**Comment by shainer** [comment]
@jeremycline thanks for the quick reply! I'll try and prepare a PR to remove the RPM dependency in the next two weeks or so.

For the Google login I might give it a try later, but I am also not very familiar with OpenID so I cannot guarantee a result.

**Comment by jeremycline** [comment]
FYI @shainer as part of my work on #332 I'm making sure the RPM dependency is either optional or gone (I haven't decided which, yet).

**Comment by shainer** [comment]
@jeremycline great thanks :)

**Comment by ncoghlan** [comment]
https://github.com/release-monitoring/anitya/pull/461 allowed the test suite to pass on Debian (which is what Travis CI uses) by falling back to a pure Python implementation of the RPM bindings when the bindings themselves are unavailable, so that may also be sufficient for Chakra's needs.

I also filed https://github.com/release-monitoring/anitya/issues/485 to cover adding GitHub as a standard option in the login menu.

**Comment by shainer** [comment]
@ncoghlan ok I'll test this week on Chakra and let you know how it goes.

**Comment by shainer** [comment]
@ncoghlan ok I pulled master, run ```python createdb.py``` and ```python runserver.py```. My instance starts and prints a warning about emulating the RPM comparison, but then requests to the service fail with:

`
sqlalchemy.exc.OperationalError: (sqlite3.OperationalError) no such column: projects.version_scheme [SQL: 'SELECT count(*) AS count_1 \nFROM (SELECT projects.id AS projects_id, projects.name AS projects_name, projects.homepage AS projects_homepage, projects.backend AS projects_backend, projects.ecosystem_name AS projects_ecosystem_name, projects.version_url AS projects_version_url, projects.regex AS
 projects_regex, projects.version_prefix AS projects_version_prefix, projects.insecure AS projects_insecure, projects.version_scheme AS projects_version_scheme, projects.latest_version AS projects_latest_version, projects.logs AS projects_logs, projects.updated_on AS 
projects_updated_on, projects.created_on AS projects_created_on \nFROM projects ORDER BY lower(projects.name)) AS anon_1']
`

I am not sure the new changes caused this, but given that launching with virtualenv worked several times I do not know what could be the cause.

**Comment by jeremycline** [comment]
@shainer Ah, it looks like you need to run the migrations. I suspect your database already existed so ``python createdb.py`` didn't do anything, but a new column got added. We use alembic for migrations, so you can run them with ``alembic upgrade head`` using the ``alembic.ini`` in the repository root (assuming the database URL is correct in there).

**Comment by shainer** [comment]
@jeremycline thanks, I recreated the database from scratch and the simulated RPM comparison seems to work.

**Comment by Zlopez** [comment]
This should be fixed with #1139
